### ISSUE TITLE
ci: difficult to pass full test due to many intermittent failure

### ISSUE THREAD

**Issue by satoru-takeuchi** [rook/rook#6358]
**Is this a bug report or feature request?**
* Bug Report

Passing the full test is very difficult due to many intermittent failures in Ceph's suites. Here are the examples.

https://github.com/rook/rook/pull/6184
https://github.com/rook/rook/pull/5741

I investigated these failures and found that there seem to be two kinds of problems.

- Intermittent failures caused by each suite.
- Failures caused by the failure of the last suite.

Here is the result of my investigation of one of the CI failures in #6184 .

# Failed tests. The capital letters in brackets are types of failures.

- k8s v1.15.12: All passed
- k8s v1.16.13: HelmSuite(A), MultiClusterDeploySuite(B), SmokeSuite(C), UpgradeSuite(D)
- k8s v1.17.9: FlexSuite(E), HelmSuite(F), MultiClusterDeploySuite(G), SmokeSuite(C), UpgradeSuite(D)
- k8s v1.18.6: All passed
- k8s v1.18.6: FlexSuite(E), HelmSuite(E), MultiClusterDeploySuite(G), SmokeSuite(C), UpgradeSuite(D)

# My hypothesis

- "A" and "E" come from each suite.
- "B"-"D" are caused be the failure of "A".
- "F" and "G" are caused by "D",

# The list of failures

## A

HelmSuite failed with the following backtrace.

```
--- FAIL: TestCephHelmSuite (411.12s)
    ceph_installer.go:568:
                Error Trace:    ceph_installer.go:568
                                                        ceph_installer.go:522
                                                        ceph_base_deploy_test.go:188
                                                        ceph_base_deploy_test.go:178
                                                        ceph_base_deploy_test.go:161
                                                        ceph_helm_test.go:80
                                                        suite.go:102
                                                        ceph_helm_test.go:52
                Error:          Received unexpected error:
                                failed to wait for clean up jobs to complete. timed out waiting for the condition
                Test:           TestCephHelmSuite
```

## B

MultiClusterDeploySuite failed with the following backtrace.

```
--- FAIL: TestCephMultiClusterDeploySuite (17.09s)
    ceph_multi_cluster_test.go:165:
                Error Trace:    ceph_multi_cluster_test.go:165
                                                        ceph_multi_cluster_test.go:151
                                                        ceph_multi_cluster_test.go:81
                                                        suite.go:102
                                                        ceph_multi_cluster_test.go:62
                Error:          Received unexpected error:
                                Failed to start admission controllers: exit status 1
                Test:           TestCephMultiClusterDeploySuite
```

## C

SmokeSuite failed with the following backtrace.

```
--- FAIL: TestCephSmokeSuite (97.10s)
    ceph_installer.go:538:
                Error Trace:    ceph_installer.go:538
                                                        ceph_installer.go:522
                                                        ceph_base_deploy_test.go:188
                                                        ceph_base_deploy_test.go:178
                                                        ceph_base_deploy_test.go:161
                                                        ceph_smoke_test.go:99
                                                        suite.go:102
                                                        ceph_smoke_test.go:71
                Error:          Received unexpected error:
                                failed to get ceph cluster. cephclusters.ceph.rook.io "test-cluster" not found
                Test:           TestCephSmokeSuite
    ceph_installer.go:568:
                Error Trace:    ceph_installer.go:568
                                                        ceph_installer.go:522
                                                        ceph_base_deploy_test.go:188
                                                        ceph_base_deploy_test.go:178
                                                        ceph_base_deploy_test.go:161
                                                        ceph_smoke_test.go:99
                                                        suite.go:102
                                                        ceph_smoke_test.go:71
                Error:          Received unexpected error:
                                failed to wait for clean up jobs to complete. timed out waiting for the condition
                Test:           TestCephSmokeSuite
```

## D

UpgradeTest failed with no backtrace.

## E

FlexSuite failed with the following backtrace.

```
--- FAIL: TestCephFlexSuite (584.14s)
    --- PASS: TestCephFlexSuite/TestBlockStorageMountUnMountForDifferentAccessModes (163.66s)
    --- PASS: TestCephFlexSuite/TestBlockStorageMountUnMountForStatefulSets (83.21s)
    --- FAIL: TestCephFlexSuite/TestFileSystem (119.51s)
        ceph_base_file_test.go:487:
                Error Trace:    ceph_base_file_test.go:487
                                                        ceph_base_file_test.go:265
                                                        ceph_flex_test.go:113
                Error:          Expected nil, but got: &errors.errorString{s:"kubectl exec command bash failed on pod rook-ceph-tools in namespace flex-ns. Failed to run: kubectl [exec -n flex-ns rook-ceph-tools -- bash -c mount -t ceph -o mds_namespace=smoke-test-fs,name=admin,secret=$(grep key /etc/ceph/keyring | awk '{print $3}') $(grep mon_host /etc/ceph/ceph.conf | awk '{print $3}'):/ /tmp/testrook] : exit status 32"}
                Test:           TestCephFlexSuite/TestFileSystem
```

## F

HelmSuite failed with the following suite. The backtrace after "ceph_installer.go:568" is the same as "A".

```
--- FAIL: TestCephHelmSuite (850.10s)
    --- PASS: TestCephHelmSuite/TestARookInstallViaHelm (0.07s)
    --- PASS: TestCephHelmSuite/TestBlockStoreOnRookInstalledViaHelm (185.38s)
    --- FAIL: TestCephHelmSuite/TestFileStoreOnRookInstalledViaHelm (250.18s)
        ceph_base_file_test.go:350:
                Error Trace:    ceph_base_file_test.go:350
                                                        ceph_helm_test.go:104
                Error:          Received unexpected error:
                                Could Not create resource in args : [apply -f -] -- Failed to run stdin: kubectl [apply -f -] : The StorageClass "cephfs-storageclass" is invalid: * parameters: Forbidden: updates to parameters are forbidden.* provisioner: Forbidden: updates to provisioner are forbidden.
                Test:           TestCephHelmSuite/TestFileStoreOnRookInstalledViaHelm
        ceph_base_file_test.go:351:
                Error Trace:    ceph_base_file_test.go:351
                                                        ceph_helm_test.go:104
                Error:          Received unexpected error:
                                Could Not create resource in args : [apply -f -] -- Failed to run stdin: kubectl [apply -f -] : The StorageClass "cephfs-storageclass" is invalid: * parameters: Forbidden: updates to parameters are forbidden.* provisioner: Forbidden: updates to provisioner are forbidden.
                Test:           TestCephHelmSuite/TestFileStoreOnRookInstalledViaHelm
        ceph_base_file_test.go:145:
                Error Trace:    ceph_base_file_test.go:145
                                                        ceph_base_file_test.go:353
                                                        ceph_helm_test.go:104
                Error:          Should be true
                Test:           TestCephHelmSuite/TestFileStoreOnRookInstalledViaHelm
                Messages:       Make sure PVC is Bound
    --- PASS: TestCephHelmSuite/TestObjectStoreOnRookInstalledViaHelm (163.40s)
    ceph_installer.go:568:
                Error Trace:    ceph_installer.go:568
                                                        ceph_installer.go:522
                                                        ceph_base_deploy_test.go:188
                                                        ceph_helm_test.go:85
                                                        suite.go:106
                                                        suite.go:139
                                                        ceph_helm_test.go:52
                Error:          Received unexpected error:
                                failed to wait for clean up jobs to complete. timed out waiting for the condition
                Test:           TestCephHelmSuite
```

## G

MultiClusterDeploySuite failed with the following backtrace.

```
--- FAIL: TestCephMultiClusterDeploySuite (239.35s)
    ceph_multi_cluster_test.go:178:
                Error Trace:    ceph_multi_cluster_test.go:178
                                                        ceph_multi_cluster_test.go:151
                                                        ceph_multi_cluster_test.go:81
                                                        suite.go:102
                                                        ceph_multi_cluster_test.go:62
                Error:          Received unexpected error:
                                failed to create cluster mrc-n1. Giving up waiting for pods with label app=rook-ceph-osd in namespace mrc-n1
                Test:           TestCephMultiClusterDeploySuite
```

**Deviation from expected behavior:**

Some tests failed.

**Expected behavior:**

All test passed

**How to reproduce it (minimal and precise):**

Run CI with `[test full]` tag

**Comment by satoru-takeuchi** [comment]
At first, I'll try to fix A and E. If my hypothesis is correct, this issue is fixed without dealing with other problems.

**Comment by satoru-takeuchi** [comment]
Pattern "E" is fixed by #6388 . Next, I'll try to fix pattern "A".

**Comment by satoru-takeuchi** [comment]
Fuxed by #6421. 

Although "Failures caused by the failure of the last suite"  would still exist, I'd like to track this problem in another new issue. It's because the critical CI instability mentioned in this issue is fixed.
